app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesActivity.kt

```diff
@@ -112,8 +111,21 @@ class SearchRepositoriesActivity : AppCompatActivity() {
         binding.searchRepo.text.trim().let {
             if (it.isNotEmpty()) {
                 binding.list.scrollToPosition(0)
-                viewModel.searchRepo(it.toString())
-                adapter.submitList(null)
+                search(it.toString())
+                // TODO how to clear the list
+                //  might not need it because of how Paging works
+//                adapter.collectFrom(PagingData.empty())
+            }
+        }
+    }
+
+    private fun search(query: String) {
+        searchJob?.cancel()
+        searchJob = lifecycleScope.launch {
+            val result = viewModel.searchRepo(query)
```

Comment:
Nit: What's `result`? I'd inline the collect because this `result` variable doesn't mean anything